### PR TITLE
Improve string reps for snapshot debugging

### PR DESCRIPTION
**Pull request by DaveCTurner**
Various things related to snapshots appear in debug logs, but have no
useful string representation which makes it hard to follow the process.
This commit adds some missing string representations.

### CODE REVIEW

**Review by ywangd — APPROVED** [review]
LGTM

File: server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java [hunk 1]
```diff
@@ -3584,9 +3600,14 @@ synchronized boolean isEmpty() {
         }
 
         synchronized boolean assertNotQueued(Snapshot snapshot) {
-            assert snapshotsToFinalize.getOrDefault(snapshot.getRepository(), new LinkedList<>())
+            if (snapshotsToFinalize.getOrDefault(snapshot.getRepository(), new LinkedList<>())
                 .stream()
-                .noneMatch(entry -> entry.equals(snapshot)) : "Snapshot [" + snapshot + "] is still in finalization queue";
+                .anyMatch(entry -> entry.equals(snapshot))) {
```
Comment (idegtiarenko) [inline server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java:3605]: It looks like `.stream().anyMatch(entry -> entry.equals(snapshot))` could be simplified to `.contains(snapshot)`

**Review by idegtiarenko — APPROVED** [review]
